### PR TITLE
Add version information

### PR DESCRIPTION
**Pull request by SuperQ**
Add a `-version` flag and metric to make it possible to monitor the
deployed version.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
CC @FranGM

**Review by twifkak — CHANGES_REQUESTED** [review]
Thanks for the contribution! This PR was an easier one for me to review (I have more context) so I'll reply to it first.

File: cmd/amppkg/main.go [hunk 1]
```diff
@@ -67,7 +69,13 @@ func (this logIntercept) ServeHTTP(resp http.ResponseWriter, req *http.Request)
 //  - It exposes an API that allows people to sign any URL as any other URL.
 //  - It is in cleartext.
 func main() {
+	prometheus.MustRegister(version.NewCollector("amppackager"))
```
[the remark this answers is not in the document]
Comment (SuperQ) [reply cmd/amppkg/main.go:72]: This seems like it would be a separate info metric, not related to the build info.

**Comment by SuperQ** [comment]
One note, with the current `go get` and `go install` commands, the `vendor` directory is never used. It downloads the modules live anyway.

Maybe it's time to drop the inline `vendor` dir?

File: README.md [hunk 2]
```diff
@@ -43,19 +53,19 @@ own and can obtain certificates for.
...
        cd my-amp-directory
-       go install github.com/ampproject/amppackager/cmd/amppkg
+       go install -ldflags "${version_flags}" github.com/ampproject/amppackager/cmd/amppkg
        ```
 
      For Go 1.14 and Go 1.15 run:
 
        ```
-       go get -u github.com/ampproject/amppackager/cmd/amppkg
+       go get -u -ldflags "${version_flags}" github.com/ampproject/amppackager/cmd/amppkg
```
[the remark this answers is not in the document]
Comment (twifkak) [reply README.md:62]: That sounds good to me! I like the idea of simplifying these instructions.

File: cmd/amppkg/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (twifkak) [reply cmd/amppkg/main.go:72]: Yeah, feel free to mark this as resolved then. Totally optional.

**Comment by twifkak** [comment]
> One note, with the current `go get` and `go install` commands, the `vendor` directory is never used. It downloads the modules live anyway.
> 
> Maybe it's time to drop the inline `vendor` dir?

Oh no, I thought I'd gotten this right finally. My apologies.

I'm torn. I like the `vendor` dir because:
 - It makes it obvious when we're increasing our binary size too much. (e.g. I made #394 in response to seeing a big addition to `vendor` in #349.)
 - It's easy-ish to scan our new dependencies and make sure they have OSS-compatible licenses.
 - It (ideally) confirms that the dependency versions in prod are the same as those tested by our CI.

But perhaps there's a better ways of doing these? (e.g. `go.{mod,sum}` is enough for the last one.)

I dislike `vendor` because:
 - It's big and weighs down our git repo and the GitHub UI.
 - Evidently I don't know how to use it.

**Comment by SuperQ** [comment]
Yea, we finally gave up on the `vendor` dir in Promteheus. Go mod is good enough for most builds.

For point 1:  Yes, it does help with that. But you need to enforce a `go mod vendor && git diff --exit-code` to make sure that all PRs correctly update vendoring.

For point 2: For dependency scanning, you can call `go mod vendor` in CI. Works quite well.

For point 3: Go modules take care of this.

We have a pretty complex Makefile in Prometheus to take care of all the vendor stuff. It's partly complex because the vendoring method changed over the years. Today we've settled on using go.mod with no vendor dir. It's good enough.

**Comment by SuperQ** [comment]
Ok, I've added a Makefile that will include the `-mod=vendor` flag if it's needed.

File: README.md [hunk 3]
```diff
@@ -31,34 +31,16 @@ own and can obtain certificates for.
...
-
-     For Go 1.13 and earlier versions run:
-
-       ```
-       go get -u -mod=vendor github.com/ampproject/amppackager/cmd/amppkg
+       make build
        ```
 
-  1. Optionally, move the built `~/go/bin/amppkg` wherever you like.
+  1. Optionally, move the built `amppkg` wherever you like.
```
Comment (twifkak) [inline README.md:43]: Yeah nice to see this whole section simplified.

**Review by twifkak — COMMENTED** [review]
Thanks! One optional change before I merge. Feel free to punt and I'll merge as is.

**Comment by twifkak** [comment]
Created #549 for removing the vendor dir.

**Comment by twifkak** [comment]
Addresses #357.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,27 @@
+BRANCH    ?= $(shell git rev-parse --abbrev-ref HEAD)
```
[the remark this answers is not in the document]
Comment (SuperQ) [reply Makefile:1]: I'm not sure what you mean? This gets the branch automatically, and isn't dependent on the workflow.

[the remark this answers is not in the document]
Comment (SuperQ) [reply Makefile:1]: Oh, you mean calling make in the workflow.

**Review by twifkak — APPROVED** [review]
Thanks! LGTM and the new actions succeed. Merging.

**Pull request merged**